delete.ps1

```diff
@@ -0,0 +1,162 @@
+    #endregion Import CSV data
+
+    if ($actionContext.CorrelationConfiguration.Enabled -eq $true) {
+        #region Get current row for person
+        $actionMessage = "querying CSV row where [$($correlationField)] = [$($correlationValue)]"
+       
+        $currentRow = $null
+        if ($csvContentGrouped -ne $null) {
+            $currentRow = $csvContentGrouped["$($correlationValue)"]
+        }
+
+        Write-Verbose "Queried CSV row where [$($correlationField)] = [$($correlationValue)]. Result count: $(($currentRow | Measure-Object).Count)"
+        #endregion Get current row for person
+    }
+
+    #region Account
+    #region Calulate action
+    $actionMessage = "calculating action"
+    if (($currentRow | Measure-Object).count -eq 1) {
+        $action = "Delete"
+    }
+    elseif (($currentRow | Measure-Object).count -gt 1) {
+        $action = "MultipleFound"
+    }
+    elseif (($currentRow | Measure-Object).count -eq 0) {
+        $action = "NotFound"
+    }
+    #endregion Calulate action
+
+    #region Process
+    switch ($action) {
+        "Delete" {
+            #region Delete csv row
+            $actionMessage = "deleting row from CSV"
+
+            #region Create custom updated csv object without current row for person
+            $updatedCsvContent = $null
+            $updatedCsvContent = [System.Collections.ArrayList](, ($csvContent | Where-Object { $_ -notin $currentRow }))
+            #endregion Create custom updated csv object without current row for person
+
+            #region Export updated CSV object
+            $exportCsvSplatParams = @{
+                Path              = $actionContext.Configuration.CsvPath
+                Delimiter         = $actionContext.Configuration.Delimiter
+                Encoding          = $actionContext.Configuration.Encoding
+                NoTypeInformation = $true
+                ErrorAction       = "Stop"
+                Verbose           = $false
+            }
+
+            if (-Not($actionContext.DryRun -eq $true)) {
+                Write-Verbose "SplatParams: $($exportCsvSplatParams | ConvertTo-Json)"
+
+                $updatedCsv = $updatedCsvContent | Foreach-Object { $_ } | Export-Csv @exportCsvSplatParams
+
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = "Deleted row from CSV [$($exportCsvSplatParams.Path)] with AccountReference: $($outputContext.AccountReference | ConvertTo-Json)."
+                        IsError = $false
+                    })
+            }
+            else {
+                Write-Warning "DryRun: Would delete row from CSV [$($exportCsvSplatParams.Path)] where [$($correlationField)] = [$($correlationValue)]."
+            }
+            #endregion Delete csv row
+
+            break
+        }
+
+        "MultipleFound" {
+            #region Multiple accounts found
+            $actionMessage = "deleting row from CSV"
```

Comment:
Seems we don't actually delete anything in this instance. So, the `$actionMessage` should be the message on line 120.